depthai_examples/ros1_src/stereo_inertial_publisher.cpp

```diff
@@ -374,32 +376,57 @@ int main(int argc, char** argv) {
                                                         nnPath);
 
     std::shared_ptr<dai::Device> device;
-    std::vector<dai::DeviceInfo> availableDevices = dai::Device::getAllAvailableDevices();
-
-    std::cout << "Listing available devices..." << std::endl;
-    for(auto deviceInfo : availableDevices) {
-        std::cout << "Device Mx ID: " << deviceInfo.getMxId() << std::endl;
-        if(deviceInfo.getMxId() == mxId) {
-            if(deviceInfo.state == X_LINK_UNBOOTED || deviceInfo.state == X_LINK_BOOTLOADER) {
-                isDeviceFound = true;
-                if(poeMode) {
-                    device = std::make_shared<dai::Device>(pipeline, deviceInfo);
-                } else {
-                    device = std::make_shared<dai::Device>(pipeline, deviceInfo, usb2Mode);
-                }
-                break;
-            } else if(deviceInfo.state == X_LINK_BOOTED) {
-                throw std::runtime_error("ros::NodeHandle() from Node \"" + pnh.getNamespace() + "\" DepthAI Device with MxId  \"" + mxId
-                                         + "\" is already booted on different process.  \"");
+    if(useWithIP) // Connecting to a camera with specific IP, requires static IP config beforehand
+    {
+        auto deviceInfo = dai::DeviceInfo(ipAddress);
+        if(deviceInfo.state == X_LINK_ANY_STATE || deviceInfo.state == X_LINK_UNBOOTED || deviceInfo.state == X_LINK_BOOTLOADER || deviceInfo.state == X_LINK_FLASH_BOOTED) {
+            std::cout << "Device found with IP Address: " << ipAddress <<  std::endl;
+            if(poeMode) {
+                device = std::make_shared<dai::Device>(pipeline, deviceInfo);
+            } else {
+                device = std::make_shared<dai::Device>(pipeline, deviceInfo, usb2Mode);
+            }
+        } else if(deviceInfo.state == X_LINK_BOOTED) {
+            throw std::runtime_error("ros::NodeHandle() from Node \"" + pnh.getNamespace() + "\" DepthAI Device with ipAddress  \"" + ipAddress
+                                     + "\" is already booted on different process.  \"");
+        } else {
+            throw std::runtime_error("Could NOT connect to device with IP Address: " + ipAddress);
+        }    
+    }
+    else if(useWithMxId) // Connecting to a camera with unique MxID
+    {
+        auto deviceInfo = dai::DeviceInfo(mxId);
+        if(deviceInfo.state == X_LINK_ANY_STATE || deviceInfo.state == X_LINK_UNBOOTED || deviceInfo.state == X_LINK_BOOTLOADER || deviceInfo.state == X_LINK_FLASH_BOOTED) {
```

Comment:
No need for Flashed_Boot here. it means device is running on it's own using the NOR flash config already.

Reply:
I will test this if it works fine, then I will remove flash booted, but as far as I remember, flashed booted comes up once you update the bootloader.

Reply:
thoughts ? @themarpe

Reply:
@samialperen 
Just updating the bootloader should not cause a flash booted state as described in the other comment. I assume you have a pipeline flashed on the device, which is then booted.

Also this code isn't valid. DeviceInfo is just a hokder of information that other functions fill out. Setting mxid to constructor doesn't fill out the state or any other fields.

Use Device.getX device to search for devices and return deviceinfo objects